src/prommis/uky/uky_flowsheet.py

```diff
@@ -1739,53 +1748,75 @@ def initialize_system(m):
-        propagate_state(m.fs.sol_feed)
-        propagate_state(m.fs.org_feed)
-        propagate_state(m.fs.org_feed2)
-
-        if stream == m.fs.leach_liquid_feed:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.leach_liquid_feed)
-        elif stream == m.fs.leach_solid_feed:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.leach_solid_feed)
-        elif stream == m.fs.rougher_org_make_up:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.rougher_org_make_up)
-        elif stream == m.fs.acid_feed1:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.acid_feed1)
-        elif stream == m.fs.acid_feed2:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.acid_feed2)
-        elif stream == m.fs.acid_feed3:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.acid_feed3)
-        elif stream == m.fs.cleaner_org_make_up:
-            print(f"Initializing {stream}")
-            initializer_feed.initialize(m.fs.cleaner_org_make_up)
-        elif stream == m.fs.leach_filter_cake:
-            print(f"Initializing {stream}")
-            initializer_product.initialize(m.fs.leach_filter_cake)
-        elif stream == m.fs.load_sep:
-            print(f"Initializing {stream}")
-            initializer_sep.initialize(m.fs.load_sep)
-        elif stream == m.fs.scrub_sep:
-            print(f"Initializing {stream}")
-            initializer_sep.initialize(m.fs.scrub_sep)
-        elif stream == m.fs.leach:
-            print(f"Initializing {stream}")
+    initializer_feed = FeedInitializer()
+    feed_units = [
+        m.fs.leach_liquid_feed,
+        m.fs.leach_solid_feed,
+        m.fs.rougher_org_make_up,
+        m.fs.acid_feed1,
+        m.fs.acid_feed2,
+        m.fs.acid_feed3,
+        m.fs.cleaner_org_make_up,
+    ]
+
+    initializer_product = ProductInitializer()
+    product_units = [
+        m.fs.leach_filter_cake,
+        m.fs.leach_filter_cake_liquid,
+        m.fs.cleaner_purge,
+        m.fs.sc_circuit_purge,
+        m.fs.precip_purge,
+    ]
+
+    initializer_sep = SeparatorInitializer()
+    sep_units = [
+        m.fs.load_sep,
+        m.fs.scrub_sep,
+        m.fs.precip_sep,
+        m.fs.cleaner_sep,
+        m.fs.rougher_sep,
+    ]
+
+    initializer_mix = MixerInitializer()
+    mix_units = [
+        m.fs.precip_sx_mixer,
+        m.fs.cleaner_mixer,
+        m.fs.rougher_mixer,
+    ]
+
+    initializer_leach = LeachingTrainInitializer()
+    leach_units = [
+        m.fs.leach,
+    ]
+
+    initializer_bt = BlockTriangularizationInitializer()
+    bt_units = [
+        m.fs.leach_mixer,
+        m.fs.leach_sx_mixer,
+        m.fs.sl_sep1,
+        m.fs.roaster,
+    ]
+
+    def function(unit):
+        if unit in feed_units:
+            print(f"Initializing {unit}")
```

Comment:
For consistency, I suggest using the logger instead of `print` for these so users can control how much output they want to see.